packages/frontend/src/app/paginated-list/paginated-list.component.spec.ts

```diff
@@ -12,123 +12,287 @@
 // See the License for the specific language governing permissions and
 // limitations under the License.
 
-import {PaginatedListComponent} from './paginated-list.component';
-import {Repository} from '../services/search/interfaces';
+import {PaginatedListComponent, PageData} from './paginated-list.component';
+import {Filter} from '../services/search/interfaces';
 import {PageEvent} from '@angular/material/paginator';
-import {TestBed, async, ComponentFixture} from '@angular/core/testing';
+import {
+  TestBed,
+  async,
+  ComponentFixture,
+  fakeAsync,
+  tick,
+} from '@angular/core/testing';
 import {environment} from 'src/environments/environment';
-import {HttpClientModule} from '@angular/common/http';
 import {MatExpansionModule} from '@angular/material/expansion';
 import {AppRoutingModule} from '../routing/app-routing.module';
 import {MatSnackBarModule} from '@angular/material/snack-bar';
+import {Observable, of, empty, throwError} from 'rxjs';
+import {COMService} from '../services/com/com.service';
 
 // Extend to use an instance
-class ImpPaginatedListComponent extends PaginatedListComponent<Repository> {}
+class ImpPaginatedListComponent extends PaginatedListComponent<PageData> {
+  fetchPageData(): Observable<PageData> {
+    return empty();
+  }
+  getEmptyElement(): PageData {
+    return {} as PageData;
+  }
+}
 
 describe('PaginatedListComponent', () => {
   let component: ImpPaginatedListComponent;
   let fixture: ComponentFixture<ImpPaginatedListComponent>;
 
+  const COMServiceMock = {};
+
   beforeEach(async(() => {
     TestBed.configureTestingModule({
       declarations: [ImpPaginatedListComponent],
-      imports: [
-        AppRoutingModule,
-        HttpClientModule,
-        MatSnackBarModule,
-        MatExpansionModule,
-      ],
+      providers: [{provide: COMService, useValue: COMServiceMock}],
+      imports: [AppRoutingModule, MatSnackBarModule, MatExpansionModule],
     }).compileComponents();
   }));
 
   beforeEach(() => {
     fixture = TestBed.createComponent(ImpPaginatedListComponent);
     component = fixture.componentInstance;
     fixture.autoDetectChanges(true);
+    component.paginator = {pageIndex: 0};
   });
 
   it('should create', () => {
     expect(component).toBeTruthy();
   });
 
-  it('should update the pages on initialization', () => {
-    spyOn(component, 'updatePage');
+  describe('onPageSelectionChange', () => {
+    it('should disable the paginator when a new page is selected', fakeAsync(() => {
```

Comment:
This is to prevent the user from being able to select the same "next page" multiple times.